udata/migrations/2016-06-24-rename-territories.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Update spatial granularities with French names.
+ *
+ */
+
+var datasets = db.dataset.find({
+    'spatial.granularity': 'fr/town'
+});
+
+print('Candidates: ' + datasets.count());
+
+var counter = 0;
+
+datasets.forEach(function (dataset) {
+    var result = db.dataset.update({_id: dataset._id},
+                                   {$set: {'spatial.granularity': 'fr/commune'}});
+    counter += result.nModified;
+});
+print(counter, 'datasets granularity modified for towns.');
+
+datasets = db.dataset.find({
+    'spatial.granularity': 'fr/commune'
+});
+
+counter = 0;
+
+datasets.forEach(function (dataset) {
```

Comment:
You are looking for [$elemMatch](https://docs.mongodb.com/manual/reference/operator/query/elemMatch/) and [$ operator](https://docs.mongodb.com/manual/reference/operator/update/positional/)